app/javascript/elm/src/Main.elm

```diff
@@ -509,7 +530,32 @@ update msg model =
             ( { model | selectedSession = SelectedSession.updateRange model.selectedSession measurements }, Cmd.none )
 
         UpdateIsShowingTimeRangeFilter isShown ->
-            ( { model | isShowingTimeRangeFilter = isShown }, Cmd.none )
+            let
+                overlay =
+                    if isShown then
+                        Overlay
+
+                    else
+                        NoOverlay
+            in
+            ( { model | overlay = updateOverlay overlay model }, Cmd.none )
```

Comment:
What do you think about having just one generic update case like so:
```
        ChangeOverlay overlayType onOrOff ->
            let
                overlay =
                    if onOrOff then
                        overlayType

                    else
                        NoOverlay
            in
            ( { model | overlay = overlay }, Cmd.none )

```

Then you can call it like so 

```
, Ports.updateIsHttping (ChangeOverlay OverlayWithSpinner)
, Ports.isShowingTimeRangeFilter (ChangeOverlay BasicOverlay)
```
or
```
        ClosePopup ->
            ( { model | popup = Popup.None, overlay = NoOverlay }, Cmd.none )

```

That would let us remove the duplication from update branches and make the logic reusable.